### PR TITLE
Update DoCIF and use ros:indigo baseimage

### PR DESCRIPTION
**Pull request by jgkamat**
Hopefully this might affect #201 

We'll see if the build works. There are a few hacky things in here, so don't merge this yet (still a WIP).

### CODE REVIEW

**Comment by jgkamat** [comment]
I would also try rebuilding master to see if it was an infra issue at all.

Could you turn on 'build forked PR's' in circleci? I reccomend not turning on buiild secrets though.

2017-04-07-204626_1492x459_scrot

**Comment by irapha** [comment]
Done! Thanks for the help Jay!

I did try rebuilding master. I mentioned it didn't work here: https://github.com/gtagency/buzzmobile/commit/34d006afe8483439912d9c1c890cb39a16daf99a

**Comment by irapha** [comment]
Actually I'm gonna try rebuilding again

**Comment by jgkamat** [comment]
Hey, it looks like the CI magically started working again (so it was probably a bug in circleci or a dockerimage that was fixed). This branch does have a few improvements though (such as pulling from ros:indigo instead of installing ros for every build), so you can use it if you want. Let me know if you want it cleaned up. :P

**Comment by irapha** [comment]
Yeah I saw it was fixed. Like 2 days after you last pushed to this PR I rebuilt master and it passed :)

I am fine with this PR, let me just make a comment or two and if you have some time to clean it up I'd appresh <3

File: circle.yml [hunk 1]
```diff
@@ -14,9 +15,6 @@ dependencies:
     - ./DoCIF/commands/buildbaseimage.sh
       # Actually the test step
     - ./DoCIF/commands/runtests.sh
-  cache_directories:
-      # This shouldn't actually cache anything because config.docif says not to
-    - ~/.ccache # See $CACHE_DIRECTORIES in config.docif
```
Comment (irapha) [inline circle.yml]: So we have plans to cache things eventually (#156). I understand this is not needed now, but I don't really see a reason to remove it :x

**Comment by jgkamat** [comment]
I'm pretty confused why preventing rosdep init would cause it to timeout (afaik, the other way is failing, and not doing anything). I'll take a closer look at that later.

File: install [hunk 2]
```diff
@@ -71,14 +71,14 @@ safe-apt-get install -y psmisc
...
-ln -s /usr/lib/python2.7/dist-packages/sip.so ./lib/python2.7/site-packages/
+ln -sf /usr/lib/python2.7/dist-packages/PyQt4/ ./lib/python2.7/site-packages/
+ln -sf /usr/lib/python2.7/dist-packages/sip.so ./lib/python2.7/site-packages/
 
 # ---------
 # Everything above this point has been installing various python dependencies
 # for things we do. Stuff below this is the ros build and install.
 
-sudo rosdep init # This can only be run once. Possible enhancement, check this
+sudo rosdep init || true # This can only be run once. Possible enhancement, check this
```
Comment (joshuamorton) [inline install]: Could you upgrade this to a TODO?

**Review by joshuamorton — COMMENTED** [review]
Single nit and fix the merge conflict, and LG. Any idea why the full set of tests didn't run on CI?

**Comment by jgkamat** [comment]
The tests actually did run (see the circleci build logs). Forked PR's don't get secrets (doing so would be a security risk), so that's why the individual status tokens aren't updating. 

I'll fix up the PR and merge master into it. This does not help with #192 in any way though (as far as I can tell). I'll also trash that 'bump' commit.

**Comment by joshuamorton** [comment]
Ah yeah, then LGTM.

**Pull request closed without merging**